packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/services/representations/RepresentationIdBuilder.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.services.representations;
+
+import org.springframework.stereotype.Service;
+
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+/**
+ * Used to build ids of representations.
+ *
+ * @author mcharfadi
+ */
+@Service
+public class RepresentationIdBuilder {
+
+    private static final String COMMA = ",";
+
+    private static final String CLOSING_BRACKETS = "]";
+
+    public String buildExplorerRepresentationId(List<String> expandedObjects, List<String> activatedFilters) {
+        List<String> expandedObjectIds = expandedObjects.stream()
+                .map(id -> URLEncoder.encode(id, StandardCharsets.UTF_8))
+                .toList();
+
+        List<String> activatedFilterIds = activatedFilters.stream()
+                .map(id -> URLEncoder.encode(id, StandardCharsets.UTF_8))
+                .toList();
+
+        return "explorer://?expandedIds=[" + String.join(COMMA, expandedObjectIds) + "]&activeFilterIds=[" + String.join(COMMA, activatedFilterIds) + CLOSING_BRACKETS;
+    }
+
+    public String buildSelectionRepresentationId(String treeDescriptionId, String targetObjectId, List<String> expandedObjectIds) {
+        return "selection://?treeDescriptionId=" + URLEncoder.encode(treeDescriptionId, StandardCharsets.UTF_8) + "&targetObjectId=" + targetObjectId + "&expandedIds=[" + String.join(COMMA, expandedObjectIds) + "]";
+    }
+
+    public String buildDiagramFilterRepresentationId(List<String> representation) {
```

Comment:
`representation` is a list of string? What is this parameter supposed to contain?

Reply:
I renamed for objectIds.
The format of the id is : `representationId: `diagramFilter://?objectIds=[${objectIds.join(',')}]`,`